src/System.Runtime/tests/System/DecimalTests.cs

```diff
@@ -1017,32 +1018,49 @@ public static IEnumerable<object[]> ToString_TestData()
             yield return new object[] { (decimal)-2468, "N", customFormat3, "(2*468.00)" };
         }
 
-        [Theory]
-        [MemberData(nameof(ToString_TestData))]
-        public static void ToString(decimal f, string format, IFormatProvider provider, string expected)
+        [Fact]
+        public static void Test_ToString()
         {
-            Helpers.PerformActionWithCulture(CultureInfo.InvariantCulture, () =>
+            RemoteInvoke(() =>
             {
-                bool isDefaultProvider = provider == null;
-                if (string.IsNullOrEmpty(format) || format.ToUpperInvariant() == "G")
+                foreach (var testdata in ToString_TestData())
                 {
-                    if (isDefaultProvider)
+                    CultureInfo originalCulture = CultureInfo.CurrentCulture;
+                    try
+                    {
+                        CultureInfo.CurrentCulture = CultureInfo.InvariantCulture;
```

Comment:
> CultureInfo.CurrentCulture = CultureInfo.InvariantCulture; [](start = 23, length = 59)

why we are doing this inside the loop and then reset it again in the finally block?